numba_cuda/numba/cuda/tests/cudapy/test_warp_ops.py

```diff
@@ -144,6 +147,47 @@ def test_shfl_sync_xor(self):
         compiled[1, nelem](ary, xor)
         self.assertTrue(np.all(ary == exp))
 
+    def test_shfl_sync_const_mode_val(self):
+        # Test `mode` argument is constant in shfl_sync calls.
+        # Related to https://github.com/NVIDIA/numba-cuda/pull/231
+        subtest = [
+            (use_shfl_sync_idx, 4),
+            (use_shfl_sync_up, 4),
+            (use_shfl_sync_down, 4),
+            (use_shfl_sync_xor, 16),
+        ]
+
+        args_re = r"\((.*)\)"
+        m = re.compile(args_re)
+
+        for func, value in subtest:
+            with self.subTest(func=func.__name__):
+                compiled = cuda.jit("void(int32[:], int32)")(func)
+                nelem = 32
+                ary = np.empty(nelem, dtype=np.int32)
+                compiled[1, nelem](ary, value)
+                irs = next(iter(compiled.inspect_llvm().values()))
+
+                for ir in irs.split("\n"):
+                    if "call" in ir and "llvm.nvvm.shfl.sync.i32" in ir:
+                        args = m.search(ir).group(0)
+                        arglist = args.split(",")
+                        mode_arg = arglist[1]
+                        self.assertNotIn("%", mode_arg)
+
+    def test_shfl_sync_const_mode_val_sm100(self):
+        # Test shfl_sync compiles with cc=(10, 0)
+        subtest = [
+            use_shfl_sync_idx,
+            use_shfl_sync_up,
+            use_shfl_sync_down,
+            use_shfl_sync_xor,
+        ]
+
+        for func in subtest:
+            with self.subTest(func=func.__name__):
+                compile_ptx(func, (int32[:], int32), cc=(10, 0))
```

Comment:
I'm a bit surprised this passed on CI on toolkits that pre-date CC 10.0. I'm looking into what's going on here now, but do you know why this didn't fail on older toolkits?

Reply:
OK, I now recall it's a long-standing behaviour that if we ask for a higher CC than supported, we just use the highest supported CC (because PTX for an older CC can be compiled for hardware of a newer CC). 

https://github.com/NVIDIA/numba-cuda/blob/9c727b0cb66e0a4b907cc27fe0da206e745436ef/numba_cuda/numba/cuda/cudadrv/nvvm.py#L478-L479